plugin/keyring_vault/i_vault_curl.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MYSQL_I_VAULT_CURL
+#define MYSQL_I_VAULT_CURL
+
+#include "i_keyring_key.h"
+#include "vault_credentials.h"
+
+namespace keyring {
+
+class IVault_curl : public Keyring_alloc
+{
+public:
+  virtual my_bool init(Vault_credentials *vault_credentials) = 0;
```

Comment:
Done, apart from const reference to Vault_credentials. Vault_credentials is just a typedef over std::map. std::map does not have const operator[] - which I am using - thus I cannot make it const.